planner/core/logical_plan_builder.go

```diff
@@ -4912,14 +4914,20 @@ func (b *PlanBuilder) BuildDataSourceFromView(ctx context.Context, dbName model.
 			terror.ErrorNotEqual(err, ErrNoSuchTable) &&
 			terror.ErrorNotEqual(err, ErrInternal) &&
 			terror.ErrorNotEqual(err, ErrFieldNotInGroupBy) &&
-			terror.ErrorNotEqual(err, ErrMixOfGroupFuncAndFields) {
+			terror.ErrorNotEqual(err, ErrMixOfGroupFuncAndFields) &&
+			terror.ErrorNotEqual(err, ErrViewNoExplain) {
 			err = ErrViewInvalid.GenWithStackByArgs(dbName.O, tableInfo.Name.O)
 		}
 		return nil, err
 	}
 
 	if tableInfo.View.Security == model.SecurityDefiner {
 		if pm := privilege.GetPrivilegeManager(b.ctx); pm != nil {
+			if viewDepth != 0 &&
+				b.ctx.GetSessionVars().StmtCtx.InExplainStmt &&
+				!pm.RequestVerification(b.ctx.GetSessionVars().ActiveRoles, dbName.L, tableInfo.Name.L, "", mysql.SelectPriv) {
```

Comment:
Why only do checks when `tableInfo.View.Security == model.SecurityDefiner` and the inner phisical table is not checked yet